databuilder/query_model.py

```diff
@@ -294,7 +294,7 @@ class In(Series[bool]):
 
 class Categorise(Series[T]):
     categories: Mapping[Series[T], Series[bool]]
-    default: Series[T]
+    default: Series[Optional[T]]
```

Comment:
I did, but thinking about it more I'm not sure I made the right choice here. Really, if we going to hint series as containing nullable things then we need to go the whole hog and do that everywhere e.g (as per our discussion earlier) we'd have `Exists(Series[bool])` and `Count(Series[int])` because they always have a value for every patient, but `Max(Series[Optional[T]])` because it can be NULL in some cases.

But I don't really want to go down that road (at least not right now) so I think the better way to model this is that the `default` argument is optional and that supplying `Value(None)` as the default is a type error because it won't match the type of the categories.

Reply:
I think that sounds right. Isn't `Value(None)` effectively what you'll get if you don't provide a `default`, though?

Reply:
Yes, it is. Which is why I don't think `Series[Optional[T]]` is strictly wrong. But it opens up the question of why this particular series gets explicit null treatment. And I'd rather people didn't start asking that question!